Sources/Swiftfall/Swiftfall.swift

```diff
@@ -1,635 +1,637 @@
-        
-        let url = URL(string: "\(scryfall)\(call)")
-        let task = URLSession.shared.dataTask(with: url!) {(data, response, error) in
-            
-            guard let content = data else {
-                print("Error: There was no data returned from JSON file.")
-                return
-            }
-    
-            //print("\(String(data: content,encoding: .utf8))")
-            
-            let decoder = JSONDecoder()
-            do {
-                // Decode JSON file starting from Response struct.
-                let decoded:Card = try decoder.decode(Card.self, from: content)
-                completion(decoded)
-            }
-            catch {
-                // Known Issues:
-                //  * Too broad of a request (needs handler) 
-                //
-                // Present an alert if the JSON data cannot be decoded.
-                do {
-                    let decoded:Error = try decoder.decode(Error.self, from: content)
-                    decoded.simplePrint()
-                }
-                catch {
-                    print("Error: \(error)")
-                    completion(nil)
-                }
-                completion(nil)
-            }
-        }
-        task.resume()
-    }
-    
-    // set
-    public static func getSet(code: String) -> Set?
-    {
-        let encodeExactly = code.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)!
-        let call = "sets/\(encodeExactly)"
-        
-        var set: Set?
-        var stop = false
-        parseCard(call: call){
-            (newset:Set?) in
-            set = newset
-            stop = true
-        }
-        
-        while(!stop){
-            //Do this until parseCard is done
-        }
-        
-        return set
-    }
-    
-    /// Retreives JSON data from URL and parses it with JSON decoder. Thanks Mitchell
-    static func parseCard(call:String, completion: @escaping (Set?) -> ()) {
-        
-        let url = URL(string: "\(scryfall)\(call)")
-        let task = URLSession.shared.dataTask(with: url!) {(data, response, error) in
-            
-            guard let content = data else {
-                print("Error: There was no data returned from JSON file.")
-                return
-            }
-            
-            //print("\(String(data: content,encoding: .utf8))")
-            
-            let decoder = JSONDecoder()
-            do {
-                // Decode JSON file starting from Response struct.
-                let decoded:Set = try decoder.decode(Set.self, from: content)
-                completion(decoded)
-            }
-            catch {
-                // Known Issues:
-                //  * Too broad of a request (needs handler)
-                //
-                // Present an alert if the JSON data cannot be decoded.
-                do {
-                    let decoded:Error = try decoder.decode(Error.self, from: content)
```

Comment:
When you are vending an API, it is _never enough_ to just print the error to the console. As a user of this API, I will not be able to access the content of this error -- if I was making an app using this, where would the error go?

You should look into using something like the `Result<T>` enum, which works like `Optional<T>`.

A good implementation is here, and you can take it as a dependency: https://github.com/antitypical/Result